File: src/main/java/net/tenakt/network/VoiceDestroyPayload.java
package net.tenakt.network;

import net.minecraft.network.RegistryByteBuf;
import net.minecraft.network.codec.PacketCodec;
import net.minecraft.network.codec.PacketCodecs;
import net.minecraft.network.packet.CustomPayload;
import net.minecraft.util.Identifier;

public record VoiceDestroyPayload(String blockId) implements CustomPayload {

    public static final CustomPayload.Id<VoiceDestroyPayload> ID = new CustomPayload.Id<>(Identifier.of("chunk-destroyer", "voice_destroy"));

    // Учим майнкрафт кодировать и раскодировать строку (название блока)
    public static final PacketCodec<RegistryByteBuf, VoiceDestroyPayload> CODEC = PacketCodec.tuple(
            PacketCodecs.STRING, VoiceDestroyPayload::blockId,
            VoiceDestroyPayload::new
    );

    @Override
    public CustomPayload.Id<? extends CustomPayload> getId() {
        return ID;
    }
}

File: src/main/kotlin/net/tenakt/network/VoiceDestroyPayload.kt
package net.tenakt.network

import net.minecraft.network.RegistryByteBuf
import net.minecraft.network.codec.PacketCodec
import net.minecraft.network.codec.PacketCodecs
import net.minecraft.network.packet.CustomPayload
import net.minecraft.util.Identifier

data class VoiceDestroyPayload(val blockId: String) : CustomPayload {
    override fun getId(): CustomPayload.Id<VoiceDestroyPayload> = ID

    companion object {
        val ID = CustomPayload.Id<VoiceDestroyPayload>(Identifier.of("chunk-destroyer", "voice_destroy"))
        val CODEC: PacketCodec<RegistryByteBuf, VoiceDestroyPayload> = PacketCodec.tuple(
            PacketCodecs.STRING, VoiceDestroyPayload::blockId,
            ::VoiceDestroyPayload
        )
    }
}